package org.apache.commons.failure.tests;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

import org.apache.commons.failure.Success;
import org.apache.commons.failure.Failure;
import org.apache.commons.failure.FailureValue;
import org.apache.commons.failure.FailureException;

import java.lang.reflect.Constructor;

class Tests {
    @Test
    void success_evaluation_test() {
    	assertEquals(new Success<String>("Test string").eval(), true);
    }
    @Test
    void failure_evaluation_test() {
    	Class<?> cls = Class.forName("org.apache.commons.failure.Failure", true, null);    
        Constructor<?> obj = cls.getConstructor();
        try {
        } catch(InstantiationException ie) {
        	
        }
    }
    @Test
    void failure_value_evaluation_test() {
    	assertEquals(new FailureValue<String>(6).eval(), false);
    }
    @Test
    void failure_exception_evaluation_test() {
    	assertEquals(new FailureException<String>(new Exception("Test exception")).eval(), false);
    }
}